### PR TITLE
Fix bug when mypy plugin fail on construct method call

### LINKED ISSUE #2753: Mypy plugin nags about "Missing named argument __config__" on BaseSettings.construct()
### Checks

* [X] I added a descriptive title to this issue
* [X] I have searched (google, github) for similar issues and couldn't find anything
* [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.8.1
            pydantic compiled: True
                 install path: /home/musttu/.pyenv/versions/3.9.4/envs/py39/lib/python3.9/site-packages/pydantic
               python version: 3.9.4 (default, May  3 2021, 13:52:18)  [GCC 9.3.0]
                     platform: Linux-5.8.0-50-generic-x86_64-with-glibc2.31
     optional deps. installed: ['typing-extensions']
```

With `mypy.ini`:

```ini
[mypy]
plugins = pydantic.mypy
```

```py
import pydantic

class Foo(pydantic.BaseSettings):
    pass

Foo.construct()
```

```
$ mypy models.py
models.py:6: error: Missing named argument "__config__" for "construct" of "Foo"
Found 1 error in 1 file (checked 1 source file)
```

Only occurs if pydntic's mypy plugin is enabled. Why is that?

### PR DESCRIPTION
**Pull request by uriyyo**
## Fix `mypy` plugin bug
Fix bug when `mypy` plugin fails on `construct` method call for `BaseSettings` derived classes.

## Related issue number
Closes #2753

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by uriyyo** [comment]
please review

File: pydantic/env_settings.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def customise_sources(
         ) -> Tuple[SettingsSourceCallable, ...]:
             return init_settings, env_settings, file_secret_settings
 
-    __config__: Config  # type: ignore
+    __config__: ClassVar[Config]  # type: ignore
```
Comment (PrettyWood) [inline pydantic/env_settings.py:116]: I just reviewed https://github.com/samuelcolvin/pydantic/pull/2721, which already fixes this (in fact it's `ClassVar[Type[Config]]`.

**Review by PrettyWood — CHANGES_REQUESTED** [review]
Please update

File: pydantic/env_settings.py (hunk 1 above)
Comment (uriyyo) [reply pydantic/env_settings.py]: Done, type annotation have been updated.

**Comment by uriyyo** [comment]
@PrettyWood Could you review again?

File: changes/2753-uriyyo.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix bug when `mypy` plugin fail on `construct` method call for `BaseSettings` derived classes.
```
Comment (PrettyWood) [inline changes/2753-uriyyo.md:1]:
```suggestion
Fix bug when `mypy` plugin fails on `construct` method call for `BaseSettings` derived classes.
```

**Review by PrettyWood — APPROVED** [review]
Thanks 🙏

**Pull request merged**